test/test_settings.py

```diff
@@ -0,0 +1,21 @@
+import festim as F
+import numpy as np
+import pytest
+
+
+@pytest.mark.parametrize("value", [1, F.Stepsize(initial_value=23.0), 1.0, "coucou"])
+def test_stepsize_value(value):
+    """Test that the stepsize is correctly set"""
+    my_model = F.HydrogenTransportProblem()
+    my_model.settings = F.Settings(atol=1, rtol=0.1)
+
+    if isinstance(value, (int, float)):
+        my_model.settings.stepsize = value
+        assert isinstance(my_model.settings.stepsize, F.Stepsize)
+        assert np.isclose(my_model.settings.stepsize.initial_value, value)
+    elif isinstance(value, F.Stepsize):
+        my_model.settings.stepsize = value
+        assert np.isclose(my_model.settings.stepsize.initial_value, 23)
+    else:
+        with pytest.raises(TypeError):
+            my_model.settings.stepsize = value
```

Comment:
I would split this test in two, and simplify it. You don't need to create a `HydrogenTransportProblem` to test the setter of `Settings`
```suggestion
@pytest.mark.parametrize("test_type", [int, F.Stepsize, float])
def test_stepsize_value(test_type):
    """Test that the stepsize is correctly set"""
    test_value = 23.0

    my_settings = F.Settings(atol=1, rtol=0.1)

    my_settings.stepsize = test_type(test_value)
    assert isinstance(my_settings.stepsize, F.Stepsize)
    assert np.isclose(my_settings.stepsize.initial_value, test_value)

def test_stepsize_value_wrong_type():
    """Checks that an error is raised when the wrong type is given"""
    my_settings = F.Settings(atol=1, rtol=0.1)
    with pytest.raises(TypeError):

        # cheese isn't a valid stepsize I'm afraid
        my_settings.stepsize = 🧀
```